src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -860,10 +860,16 @@ public SnapshotPreviewModel retrievePreview(
     }
   }
 
-  private String getOrCreateExternalDataSource(
-      AuthenticatedUserRequest userRequest, String tableName, Snapshot snapshot) {
+  /**
+   * @param snapshot the snapshot to configure the AzureDataSourceFor
+   * @param userRequest the user making the request
+   * @return the name of the datasource created
+   * @throws RuntimeException when the external datasource could not be configured
+   */
+  public String getOrCreateExternalAzureDataSource(
+      Snapshot snapshot, AuthenticatedUserRequest userRequest) {
     AccessInfoModel accessInfoModel =
-        metadataDataAccessUtils.accessInfoFromSnapshot(snapshot, userRequest, tableName);
```

Comment:
Thanks for asking about this -- We don't technically need to pass in the table name. If it is not specified, then it will retrieve access information for all of the tables. However, if we only need info for one table, it is much faster to do it just for the single table. 
We need the more general case with all of the tables for the SnapshotBuilderService.createContext() method. But, we can continue to use the more specific one for the call above in the snapshot service.